### PR TITLE
Improve Databricks connector docs / add `delta_lake` docs

### PR DESCRIPTION
**Pull request by phillipleblanc**
## 🗣 Description

Improves the Databricks connector docs based on the enhancements in https://github.com/spiceai/spiceai/issues/1696 and adds docs for the new `delta_lake` connector.

## 🔨 Related Issues

https://github.com/spiceai/spiceai/issues/1696

### CODE REVIEW

File: spiceaidocs/docs/data-connectors/databricks.md [hunk 1]
```diff
@@ -8,21 +8,71 @@ pagination_prev: null
 import Tabs from '@theme/Tabs';
 import TabItem from '@theme/TabItem';
 
-Databricks as a connector for federated SQL query against Databricks using [Spark Connect](https://www.databricks.com/blog/2022/07/07/introducing-spark-connect-the-power-of-apache-spark-everywhere.html) or directly from Delta Tables in S3.
+Databricks as a connector for federated SQL query against Databricks using [Spark Connect](https://www.databricks.com/blog/2022/07/07/introducing-spark-connect-the-power-of-apache-spark-everywhere.html) or directly from [Delta Lake](https://delta.io/) Tables.
```
Comment (digadeesh) [inline spiceaidocs/docs/data-connectors/databricks.md]:
```suggestion
Databricks as a connector for federated SQL query against Databricks using [Spark Connect](https://www.databricks.com/blog/2022/07/07/introducing-spark-connect-the-power-of-apache-spark-everywhere.html) or directly from [Delta Lake](https://delta.io/) tables.
```

File: spiceaidocs/docs/data-connectors/databricks.md [hunk 2]
```diff
@@ -114,9 +164,17 @@ Check [Secrets Stores](/secret-stores) for more details.
 
 ```yaml
 datasets:
+  # Example for Spark Connect
+  - from: databricks:spiceai.datasets.my_awesome_table  // A reference to a table in the Databricks unity catalog
+    name: my_delta_lake_table
+    params:
+      endpoint: dbc-a1b2345c-d6e7.cloud.databricks.com
+      mode: spark_connect
+      databricks-cluster-id: 1234-567890-abcde123
```
Comment (digadeesh) [inline spiceaidocs/docs/data-connectors/databricks.md]:
```suggestion
      databricks_cluster_id: 1234-567890-abcde123
```

**Review by digadeesh — DISMISSED** [review]

File: spiceaidocs/docs/data-connectors/delta-lake.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+  </TabItem>
+</Tabs>
+
+## Example
+
+```yaml
+datasets:
+  # Example for local Delta Lake
+  - from: delta_lake:/path/to/local/delta/table  // A local filesystem path to a Delta Lake table
```
Comment (sgrebnov) [inline spiceaidocs/docs/data-connectors/delta-lake.md:43]: `delta_lake` connector should be added to index page as well: https://github.com/spiceai/docs/blob/trunk/spiceaidocs/docs/data-connectors/index.md

Comment (sgrebnov) [reply spiceaidocs/docs/data-connectors/delta-lake.md:43]: PR to add https://github.com/spiceai/docs/pull/308

**Review by sgrebnov — DISMISSED** [review]

**Review by digadeesh — APPROVED** [review]

**Pull request merged**